flink-connector-bigquery/src/test/java/com/google/cloud/flink/bigquery/sink/serializer/AvroToProtoSerializerTestSchemas.java

```diff
@@ -0,0 +1,252 @@
+        Descriptor descriptor = bigQuerySchemaProvider.getDescriptor();
+        return new BigQueryAvroToProtoSerializerTestResult(
+                bigQuerySchemaProvider.getSchema(), descriptor);
+    }
+
+    public static BigQueryAvroToProtoSerializerTestResult testLogicalTypesConversion() {
+        List<TableFieldSchema> fields =
+                Arrays.asList(
+                        new TableFieldSchema()
+                                .setName("timestamp")
+                                .setType("TIMESTAMP")
+                                .setMode("NULLABLE"),
+                        new TableFieldSchema()
+                                .setName("numeric_field")
+                                .setType("NUMERIC")
+                                .setMode("REQUIRED"),
+                        new TableFieldSchema()
+                                .setName("bignumeric_field")
+                                .setType("BIGNUMERIC")
+                                .setMode("NULLABLE"),
+                        new TableFieldSchema()
+                                .setName("geography")
+                                .setType("GEOGRAPHY")
+                                .setMode("REQUIRED"),
+                        new TableFieldSchema().setName("Json").setType("JSON").setMode("REQUIRED"));
+
+        TableSchema tableSchema = new TableSchema().setFields(fields);
+        BigQuerySchemaProvider bigQuerySchemaProvider = new BigQuerySchemaProvider(tableSchema);
+        Descriptor descriptor = bigQuerySchemaProvider.getDescriptor();
+        return new BigQueryAvroToProtoSerializerTestResult(
+                bigQuerySchemaProvider.getSchema(), descriptor);
+    }
+
+    public static BigQueryAvroToProtoSerializerTestResult testAllPrimitiveSchemaConversion() {
+        String fieldString =
+                " \"fields\": [\n"
+                        + "   {\"name\": \"name\", \"type\": \"string\"},\n"
+                        + "   {\"name\": \"number\", \"type\": \"long\"},\n"
+                        + "   {\"name\": \"quantity\", \"type\": \"int\"},\n"
+                        + "   {\"name\": \"fixed_field\", \"type\": {\"type\": "
+                        + "\"fixed\", \"size\": 10,\"name\": \"hash\" }},\n"
+                        + "   {\"name\": \"price\", \"type\": \"float\"},\n"
+                        + "   {\"name\": \"double_field\", \"type\": \"double\"},\n"
+                        + "   {\"name\": \"boolean_field\", \"type\": \"boolean\"},\n"
+                        + "   {\"name\": \"enum_field\", \"type\": {\"type\":\"enum\","
+                        + " \"symbols\": [\"A\", \"B\", \"C\", \"D\"], \"name\": \"ALPHABET\"}},\n"
+                        + "   {\"name\": \"byte_field\", \"type\": \"bytes\"}\n"
+                        + " ]\n";
+        return getSchemaAndDescriptor(fieldString);
+    }
+
+    public static BigQueryAvroToProtoSerializerTestResult testAllLogicalSchemaConversion() {
```

Comment:
How will this be used differently from the logical schema provider above?

Reply:
UUID, Time/lts and ts millis are not obtained when converting from BQ table schema. Removed all other redundant fields.